components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -142,10 +151,27 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
         if (valid) {
             //Add authorized internal scopes to the request for sending in the response.
             addAuthorizedInternalScopes(authzReqMsgCtx, authorizedInternalScopes);
+            addAllowedScopes(authzReqMsgCtx, requestedAllowedScopes.toArray(new String[requestedAllowedScopes.size()]));
         }
         return authorizeRespDTO;
     }
 
+    /**
+     * Determines if the scope is specified in the whitelist.
+     *
+     * @param scope - The scope key to check
+     * @return - 'true' if the scope is white listed. 'false' if not.
+     */
+    public boolean isAllowedScope(List<String> scopeSkipList, String scope) {
```

Comment:
Duplicate code. Reuse the method defined in OAuth2Util

Reply:
removed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1468/commits/4d2bf08c421feabe46cf80c0cda56853705228d8